modules/@apostrophecms/migration/index.js

```diff
@@ -286,14 +288,18 @@ module.exports = {
             throw err;
           }
         }
-      }
+      },
+      ...require('./lib/addMissingSchemaFields.js')(self)
     };
   },
   tasks(self) {
     return {
       migrate: {
         usage: 'Apply any necessary migrations to the database.',
-        task: self.migrate
+        // Migrations actually run on every invocation
+        // and automatically detect whether any work
+        // must be done
+        task: () => {}
```

Comment:
Previously this task invoked the migrate code for a third time 🪦

Reply:
The task must exist otherwise users will get an error when launching the migration task according to our documentation. The fact that it is implicit is an implementation detail that should not concern users. Later we may eliminate the implicit invocation except when the task is being invoked, in production.

Reply:
@haroun it has worked like this for about as long as A3 has existed. Plus extra, redundant invocations in two other places. Since we have this automatic invocation at every startup I am fixing the bug of redundant invocations. As I mentioned we might reconsider this behavior later for production, but in dev it never makes sense to start up without migrating, if your code does not behave after migrations then that is a bug in the migrations that needs attention and something that will break in production if bypassed, etc.